spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -153,6 +153,82 @@
         "order1@example.com"
       ].join(" "))
     end
+
+    context "with a voucher" do
+      let(:voucher) do
+        create(:voucher_flat_rate, code: 'some_code', enterprise: distributor, amount: 10)
+      end
+
+      before do
+        Flipper.enable :vouchers
+
+        # Add voucher to the order
+        voucher.create_adjustment(voucher.code, order)
+        VoucherAdjustmentsService.new(order).update
+
+        # the enterprise fees can be known only when the user selects the variants
+        # we'll need to create them by calling recreate_all_fees!
+        order.recreate_all_fees!
+
+        order.update_totals_and_states
+        OrderWorkflow.new(order).complete!
+      end
+
+      pending "generates the report with the correct tax amount" do
+        visit_sales_tax_totals_by_order
+
+        expect(page).to have_button("Go")
+        click_on "Go"
+        expect(page.find("table.report__table thead tr").text).to have_content(table_header)
+
+        # Calculation details :
+        # Price excl tax = order.total - total_tax + voucher tax (negative number)
+        # Tax amount = Non adjusted tax - voucher tax
+        # Price incl tac = Price excl tax + non adjusted tax
+        expect(page.find("table.report__table tbody").text).to have_content([
+          "DistributorEnterpriseWithTax",
+          "oc1",
+          "ORDER_NUMBER_1",
+          "GST Food",
+          "State",
+          "1.5 %",
+          "105.39", # 109.61 - 4.61 + 0.39
```

Comment:
I am not sure where you got your `115.00` from ? The spec above has the order total = 119.61 , so if you apply a $10 voucher  the order total becomes `109.61`.

Reply:
I think I was looking at this value: https://github.com/openfoodfoundation/openfoodnetwork/pull/11543/files/29a4bf88d7520b2e26615fa7e44a9a6a27c65ef7..a7bceb0b28a7ac863126ce59db15039bce4894a0#diff-ef69e93a7cc19d5540d0ef31a1153f01703706454194c29d2588cadffca68e11R120
But I see now that was total *excl.* tax. The total *incl.* tax is `116.73` though..

In any case, I'm thinking it would be good to arrange our tests in a way that show what the values mean. In this situation I would probably add a comment to each value. 
But I wonder if we could construct a helper to express these expections better. Something like:

```
food_row = row_with_content("DistributorEnterpriseWithTax oc1 ORDER_NUMBER_1 GST Food")
expect(food_row).to have_column_value("Total excl. Tax ($)": "115.0")
```

Hmm that's a bit wordy.. maybe I'll stick with my comment idea ;)